content/post/078-go-ipfs-0.4.23.md

```diff
@@ -0,0 +1,84 @@
+---
+date: 2020-01-27
+url: 2020-01-27-go-ipfs-0-4-23
+title: go-ipfs 0.4.23 released
+author: Alan Shaw
+---
+
+> Would sir/madam care for another patch release while they wait?
+
+Yes that's right, the next feature release of go-ipfs (0.5.0) is, well, running a tiny bit behind schedule. In the mean time though we have patches, and I'm not talking pirate eye patches, I'm talking bug fixes. We're hunting these bugs like they're Pokemon, and jeez, do we come across some rare and difficult to fix ones? - you betcha.
+
+Alright, enough funny business, what's the deal? Ok so, I don't want to _alarm_ anyone but this release has some **critical fixes** and if you're using go-ipfs or know someone who is then you and your friends need to slide into your upgrade pants and give those IPFS nodes a good wipe down ASAP.
+
+## 🔦 Highlights
+
+### 🤝 Fixed Spontaneous TLS Disconnects
+
+If this isn't reason enough to upgrade I don't know what is. Turns out, a TLS handshake _may_ have accidentially been unintentionally aborted for no good reason 😱. I shouldn't need to tell you this but \*whispers\* that's really bad for connectivity! Excellent news though, it's fixed, and if you want to stay well connected in the glorious future of p2p communications then you should definitely upgrade.
+
+### 😱 Fixed Panics and Crashes
+
+Panicing won't help, in life, and also in golang. Stay calm and breathe slowly. We patched a number of panics and crashes that were uncovered, including a panic seen commonly in the websocket transport. High ten 🙌?
+
+### 🔁 Fixed Resursive Resolving of dnsaddr Multiaddrs
+
+Imagine this: rolling out a brand spanking new set of bootstrap nodes only to discover their new addresses are not resolvable. I know right - bad news bears 🐻!?
+
+Ok, "bear" this in mind: you want to keep all your DNS TXT records [below 512 bytes to avoid UDP fragmentation](https://serverfault.com/questions/840241/do-dns-queries-always-travel-over-udp), otherwise you'll get a truncated reply and have to connect with TCP to get all the records. If you have lots of dnsaddr TXT records then it can be more efficient to use recursive resolving than to get a truncated reply and go through the famous 18-way SYN, SYN-ACK ACK, ACK-SYN, ACK-ACK (...etc, etc) TCP handshake, not to mention the fact that go-ipfs will not even try to fallback to TCP 😅.
+
+Anyway, long story short. We fixed that. You're welcome.
```

Comment:
It's a bit hard to figure out what we fixed here. We should probably explicitly mention that we fixed recursive resolution of dnsaddrs, not DNS record fragmentation.

And maybe explain what we mean by that (/dnsaddr/a.com -> /dnsaddr/foo.a.com -> ...). Or maybe not, that's up to you and may be too much detail.